tests/jobs/test_base_job.py

```diff
@@ -205,15 +205,17 @@ def test_is_alive_scheduler(self, job_type):
         assert job.is_alive() is False, "Completed jobs even with recent heartbeat should not be alive"
 
     def test_heartbeat_failed(self):
-        when = timezone.utcnow() - datetime.timedelta(seconds=60)
-        mock_session = Mock(name="MockSession")
-        mock_session.commit.side_effect = OperationalError("Force fail", {}, None)
-        job = Job(heartrate=10, state=State.RUNNING)
-        job.latest_heartbeat = when
-
-        job.heartbeat(heartbeat_callback=lambda: None, session=mock_session)
-
-        assert job.latest_heartbeat == when, "attribute not updated when heartbeat fails"
+        try:
+            when = timezone.utcnow() - datetime.timedelta(seconds=60)
+            mock_session = Mock(name="MockSession")
+            mock_session.commit.side_effect = OperationalError("Force fail", {}, None)
+            job = Job(heartrate=10, state=State.RUNNING)
+            job.latest_heartbeat = when
+            job.heartbeat(heartbeat_callback=lambda: None, session=mock_session)
+            assert job.latest_heartbeat == when, "attribute not updated when heartbeat fails"
+            assert job.heartbeat_failed
+        except OperationalError as excinfo:
```

Comment:
Why OperationalError here? I believe it is handled internally (when I remove it , the test continues to work BTW.

Reply:
Let me get back to you on this point, before this week ends @potiuk

Reply:
@potiuk I wrongly thought that OperationalError will be thrown to this code, without checking that that exception is handled and the right error messages and params are set. I have removed that part of the code.